utils/usage/usage.go

```diff
@@ -0,0 +1,225 @@
+package usage
+
+import (
+	"golang.org/x/sync/errgroup"
+
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	xray "github.com/jfrog/jfrog-cli-core/v2/xray/commands/utils"
+	"github.com/jfrog/jfrog-client-go/artifactory/usage"
+	clientutils "github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	ecosysusage "github.com/jfrog/jfrog-client-go/utils/usage"
+	xrayusage "github.com/jfrog/jfrog-client-go/xray/usage"
+)
+
+const (
+	ReportUsagePrefix     = "Usage Report:"
+	clientIdAttributeName = "clientId"
+)
+
+type UsageReporter struct {
+	ProductId       string
+	serverDetails   *config.ServerDetails
+	reportWaitGroup *errgroup.Group
+
+	sendToEcosystem   bool
+	sendToXray        bool
+	sendToArtifactory bool
+}
+
+type ReportFeature struct {
+	FeatureId  string
+	ClientId   string
+	Attributes []ReportUsageAttribute
+}
+
+type ReportUsageAttribute struct {
+	AttributeName  string
+	AttributeValue string
+}
+
+func NewUsageReporter(productId string, serverDetails *config.ServerDetails) *UsageReporter {
+	return &UsageReporter{
+		ProductId:         productId,
+		serverDetails:     serverDetails,
+		reportWaitGroup:   new(errgroup.Group),
+		sendToEcosystem:   true,
+		sendToXray:        true,
+		sendToArtifactory: true,
+	}
+}
+
+func ShouldReportUsage() (reportUsage bool) {
```

Comment:
Let's use this function also in -
```
func reportUsage(command Command, channel chan<- bool) {
	defer signalReportUsageFinished(channel)
	reportUsage, err := clientutils.GetBoolEnvValue(coreutils.ReportUsage, true)
	if err != nil {
		log.Debug(usageReporter.ReportUsagePrefix, err.Error())
		return
	}
```
instead of calling `clientutils.GetBoolEnvValue` directly.

Also, why don't we remove the above `reportUsage` function and use the new UsageReporter instead?